x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_empty_prompt.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiEmptyPrompt } from '@elastic/eui';
+import type { FunctionComponent } from 'react';
+import React from 'react';
+
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { DocLink } from '../../../components/doc_link';
+
+export interface ApiKeysEmptyPromptProps {
+  error?: Error;
+}
+
+export const ApiKeysEmptyPrompt: FunctionComponent<ApiKeysEmptyPromptProps> = ({
+  error,
+  children,
+}) => {
+  if (error) {
+    if (doesErrorIndicateAPIKeysAreDisabled(error)) {
+      return (
+        <EuiEmptyPrompt
+          iconType="alert"
+          body={
+            <>
+              <p>
+                <FormattedMessage
+                  id="xpack.security.management.apiKeysEmptyPrompt.disabledErrorMessage"
+                  defaultMessage="API keys are disabled."
+                />
+              </p>
+              <p>
+                <DocLink app="elasticsearch" doc="security-settings.html#api-key-service-settings">
+                  <FormattedMessage
+                    id="xpack.security.management.apiKeysEmptyPrompt.docsLinkText"
+                    defaultMessage="Learn how to enable API keys."
+                  />
+                </DocLink>
+              </p>
+            </>
+          }
+        />
+      );
+    }
+
+    if (doesErrorIndicateUserHasNoPermissionsToManageAPIKeys(error)) {
+      return (
+        <EuiEmptyPrompt
+          iconType="lock"
+          body={
+            <p>
+              <FormattedMessage
+                id="xpack.security.management.apiKeysEmptyPrompt.forbiddenErrorMessage"
+                defaultMessage="Not authorized to manage API keys."
+              />
+            </p>
+          }
+        />
+      );
+    }
+
+    return (
+      <EuiEmptyPrompt
+        iconType="alert"
+        body={
+          <p>
+            <FormattedMessage
+              id="xpack.security.management.apiKeysEmptyPrompt.errorMessage"
+              defaultMessage="Could not load API keys."
+            />
+          </p>
+        }
+        actions={children}
+      />
+    );
```

Comment:
The page does not show the create button when initial data fetch failed. 

I don't think we should show error codes or messages from the API unless they're intended to be displayed in the UI.

Reply:
Well, the page showed these details before, and other pages show similar details when we encounter unexpected errors. If we don't include this it will be difficult to troubleshoot in support cases.

Reply:
We chatted about this a bit offline, following up here --

To clarify, I tested this in both the current master branch and this PR by adding this bit to throw errors 50% of the time:
```diff
diff --git a/x-pack/plugins/security/server/routes/api_keys/get.ts b/x-pack/plugins/security/server/routes/api_keys/get.ts
index 7862401e1b8..0d44fe9acc3 100644
--- a/x-pack/plugins/security/server/routes/api_keys/get.ts
+++ b/x-pack/plugins/security/server/routes/api_keys/get.ts
@@ -29,6 +29,9 @@ export function defineGetApiKeysRoutes({ router }: RouteDefinitionParams) {
     },
     createLicensedRouteHandler(async (context, request, response) => {
       try {
+        if (Math.random() < 0.5) {
+          throw new Error('Test error');
+        }
         const isAdmin = request.query.isAdmin === 'true';
         const apiResponse = await context.core.elasticsearch.client.asCurrentUser.security.getApiKey<{
           api_keys: ApiKey[];
```

Error in current master:

![api-key-error](https://user-images.githubusercontent.com/5295965/113918177-19ee0680-97b0-11eb-8e91-3c920e90f6cd.gif)

Error in this PR:

![image](https://user-images.githubusercontent.com/5295965/113918033-f1660c80-97af-11eb-8861-16a0ac6d09de.png)

So it would be great if we could display the error status code and message in this callout.

Also I inadvertently discovered a bug -- if you encounter an error and click "Reload", even if the request is successful, the UI doesn't change. I tested this change and it fixed the issue:

```diff
diff --git a/x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_grid_page.tsx b/x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_grid_page.tsx
index 014030a8927..66eed00eb45 100644
--- a/x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_grid_page.tsx
+++ b/x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_grid_page.tsx
@@ -681,7 +681,7 @@ export class APIKeysGridPage extends Component<Props, State> {
     try {
       const { isAdmin } = this.state;
       const { apiKeys } = await this.props.apiKeysAPIClient.getApiKeys(isAdmin);
-      this.setState({ apiKeys });
+      this.setState({ apiKeys, error: undefined });
     } catch (e) {
       this.setState({ error: e });
     }
```